### PR TITLE
Push various javadoc generation code into jdt.core.manipulations

### PR DESCRIPTION
**Pull request by robstryker**
## Author checklist

- [x] I have thoroughly tested my changes
- [x] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [x] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by robstryker** [comment]
This PR actually depends on https://github.com/eclipse-platform/eclipse.platform.ui/pull/1334  and, once 1334 is merged, this PR will need to be modified to remove the duplicate classes. 

They were only placed in this PR so that I could have the build system run the test suite repeatedly.

File: org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/core/manipulation/internal/javadoc/JavaDocMessages.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.jdt.core.manipulation.internal.javadoc;
+
+import org.eclipse.osgi.util.NLS;
+
+/**
+ * Helper class to get NLSed messages.
+ */
+final class JavaDocMessages extends NLS {
```
Comment (robstryker) [inline org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/core/manipulation/internal/javadoc/JavaDocMessages.java:21]: Remember to delete unused strings

**Comment by rgrunber** [comment]
I can have a look probably by end of the week unless someone gets to it before me.

**Comment by jjohnstn** [comment]
> I can have a look probably by end of the week unless someone gets to it before me.

I'm reviewing it now

**Review by jjohnstn — APPROVED** [review]
Looks good.

File: org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/core/manipulation/internal/javadoc/CoreJavadocAccess.java [hunk 2]
```diff
@@ -0,0 +1,478 @@
...
+					superMethodReferences.append(attachedDocInHierarchy);
+					return superMethodReferences.toString();
+				}
+			}
+		}
+		return sourceJavadoc;
+	}
+
+	record SuperclassReferenceMethodData(ArrayList<IMethod> superInterfaceMethods, IMethod superClassMethod) {
+	};
```
[the remark this answers is not in the document]
Comment (jukzi) [reply org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/core/manipulation/internal/javadoc/CoreJavadocAccess.java:269]: not your fault. I am working on github shows that warning too, so reviewer can see it

**Pull request merged**